### PR TITLE
 csharpier: init at 0.26.4 

### PR DESCRIPTION
**Pull request by zoriya**
## Description of changes

CSharpier is an opinionated code formatter for c#. It's like prettier but for csharp.

Homepage: https://csharpier.com/
Github: https://github.com/belav/csharpier

PS: this is my first time packaging something for nix, if I did something wrong, please inform me.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/cs/csharpier/package.nix
Comment (h7x4) [inline pkgs/by-name/cs/csharpier/package.nix:14]: [Upstream seems to specify MIT license](https://github.com/belav/csharpier/blob/main/LICENSE)

```suggestion
    license = licenses.mit;
```

File: pkgs/by-name/cs/csharpier/package.nix
Comment (h7x4) [inline pkgs/by-name/cs/csharpier/package.nix:11]: Nit

```suggestion
    description = "CSharpier is an opinionated code formatter for C#";
```

File: pkgs/by-name/cs/csharpier/package.nix
Comment (h7x4) [inline pkgs/by-name/cs/csharpier/package.nix:16]:
```suggestion
    maintainers = with maintainers; [ zoriya ];
    mainProgram = "dotnet-csharpier";
```

**Review by h7x4 — COMMENTED** [review]
Here's a few comments

**Comment by emilioziniades** [comment]
Hi @zoriya, I'm bumping this because I would really love to see this merged. I think you need to request another review from @h7x4 to trigger a notification.

Left a comment about the platform, there was a similar discussion that happened on #281005 for another csharp project.

**Comment by zoriya** [comment]
Thanks, @emilioziniades. I fixed your comment and asked @h7x4 back for a review.

**Comment by emilioziniades** [comment]
Result of `nixpkgs-review pr 272806` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>csharpier</li>
  </ul>
</details>

**Comment by emilioziniades** [comment]
Result of `nixpkgs-review pr 272806` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>csharpier</li>
  </ul>
</details>

**Review by emilioziniades — APPROVED** [review]
LGTM!

File: pkgs/by-name/cs/csharpier/package.nix
Comment (h7x4) [inline pkgs/by-name/cs/csharpier/package.nix:11]: I think the description needs a small touch: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

```suggestion
    description = "An opinionated code formatter for C#";
```

File: pkgs/by-name/cs/csharpier/package.nix
Comment (h7x4) [inline pkgs/by-name/cs/csharpier/package.nix:15]: This can be dropped completely. Upstream doesn't clearly specify that they limit platforms anywhere, and `buildDotnetModule` (called through `buildDotnetGlobalTool`) handles the platforms for us.

https://github.com/NixOS/nixpkgs/blob/a5845db50b44e4cb5867babf1110d3dbe4f65687/pkgs/build-support/dotnet/build-dotnet-module/default.nix#L93-L96

```suggestion
```

File: pkgs/by-name/cs/csharpier/package.nix
Comment (h7x4) [inline pkgs/by-name/cs/csharpier/package.nix:5]: [`0.27.0`](https://github.com/belav/csharpier/releases/tag/0.27.0) seems to have just released, if you'd like to bump before merge.

**Review by h7x4 — COMMENTED** [review]
Only small things left :)

**Comment by zoriya** [comment]
Done @h7x4.

**Review by h7x4 — APPROVED** [review]

**Comment by sharpchen** [comment]
Why `dotnet-csharpier` instead of just `csharpier`?

**Pull request merged**
